src/os_posix/os_dir.c

```diff
@@ -64,22 +84,37 @@ __directory_list_worker(WT_FILE_SYSTEM *file_system, WT_SESSION *wt_session, con
         if (single)
             break;
     }
-
+    /*
+     * Reading the directory returns NULL on failure or reaching the end of the list. So set the
+     * value into ret as soon as the loop exits.
+     */
+    ret = errno;
```

Comment:
I'd add a check for a non-zero ret here and print a message if the read fails (otherwise we get the same closedir fail message).

Reply:
Added.